### PR TITLE
SQLDataSetMap

### PR DESCRIPTION
**Pull request by kaapstorm**
## Summary

Jira: [SC-1053](https://dimagi-dev.atlassian.net/browse/SC-1053)

1. Migrates DataSetMap and DataValueMap models to SQL
2. Replaces (broken) single-page UI with two CRUDPaginatedView-based pages for managing SQLDataSetMap instances and the SQLDataValueMap instances they contain.

#### Context:

* A "DataSet map" maps a UCR to a DHIS2 "DataSet". A DataSet is a set of indicators.
* A "DataValue map" maps a UCR column to a "DataValue". A DataValue is a "data element"-"category option combo" pair. An example of a data element is "Live Births", and its category option combos are the permutations of gender + birth attendants, e.g. "male + gynobs", "female + gynobs", "male + doula", ... etc.

#### Not included:

* This PR does not include using DHIS2 metadata to replace input fields for DHIS2 IDs with select dropdowns. 
* It also does not include a JSON UI for adding and editing DataSet maps.

#### DataSet Map list:
dataset_maps_1

#### Adding a new DataSet Map:
dataset_maps_2

#### Editing a DataSet Map that has DataValue Maps:
dataset_maps_3

#### Editing a DataValue Map:
dataset_maps_4

I am opening this as a draft PR for feedback on two questions:
1. I think this change should come with more tests. Currently just one test case covers `SQLDataSetMap` ([GetInfoForColumnsTests includes SQL model](https://github.com/dimagi/commcare-hq/commit/6f6f2281f392e8060fd9845928bf5b0618dc4cf2)). But I'm not sure how best to test the views. Any suggestions, or examples from the codebase where we do this well?
2. I am unsure how to namespace the `sendNow()` JavaScript function ([Add "Send Now" button](https://github.com/dimagi/commcare-hq/commit/2d22306ef284a679f6dddc3729e73141c38910a2)). Is there a better way to define and call it?

## Feature Flag
DHIS2 Integration

## Product Description
Improved user interface for DHIS2 aggregated data integration.

## Safety Assurance

- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I am certain that this PR will not introduce a regression for the reasons below

### Automated test coverage

Model changes are covered by "a non-zero number" of tests. But I'd like to add automated tests for other changes.

### QA Plan

[QA ticket](https://dimagi-dev.atlassian.net/browse/QA-2195)

### Safety story

* The number of models to be migrated is small: This change only affects projects that use DHIS2 aggregated data integration.
* Local testing has shown that adding, updating and deleting DataSet Maps and DataValue Maps is reliable, and that the "Send Now" button behaves as expected.
* Once I have follow up on feedback regarding automated tests, I will switch from a draft pull request to an actual pull request.

#### Immediate rollback

If no new `SQLDataSetMap` or `SQLDataValueMap` instances have been created since this PR was deployed, it can be reverted with no further considerations.

#### Delayed rollback

If it is possible that a DHIS2 aggregated data integration has been set up since this PR was deployed, it can be reverted in two steps:
1. Run the "populate_couch_datasetmap" management command on the deployed environment to copy new SQL models to Couch.
2. Revert the PR and deploy.

### CODE REVIEW

File: corehq/motech/dhis2/static/dhis2/js/dataset_map_create.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 hqDefine("dhis2/js/dataset_map_create", [
     "jquery",
     "hqwebapp/js/widgets",
-    'jquery-ui/ui/datepicker',
+    'jquery-ui/ui/widgets/datepicker',
```
Comment (kaapstorm) [inline corehq/motech/dhis2/static/dhis2/js/dataset_map_create.js:4]: Thanks @orangejenny !

**Comment by kaapstorm** [comment]
Dropping "Awaiting QA" ticket. QA is complete.

There is one blocking QA ticket, [QA-2680](https://dimagi-dev.atlassian.net/browse/QA-2680) (where, when two datasets with different frequency are scheduled to be sent on the same day, *sometimes* one is sent twice and the other is skipped).

The rest of the QA tickets are low priority and will be followed up after this PR and #29416 are deployed.

File: corehq/motech/dhis2/forms.py [hunk 2]
```diff
@@ -1,20 +1,367 @@
...
+        helper = FormHelper()
+        helper.form_style = 'inline'
+        helper.form_show_labels = False
+        helper.layout = crispy.Layout(
+            twbscrispy.InlineField('column'),
+            twbscrispy.InlineField('data_element_id'),
+            twbscrispy.InlineField('category_option_combo_id'),
+            twbscrispy.InlineField('comment'),
+            twbscrispy.StrictButton(
+                mark_safe(f'<i class="fa fa-plus"></i> {_("Add")}'),
```
Comment (mjriley) [inline corehq/motech/dhis2/forms.py:315]: Can you add a comment to hide this (safe) usage of `mark_safe` from automated checks?

```suggestion
                mark_safe(f'<i class="fa fa-plus"></i> {_("Add")}'),  # nosec: no user input
```

**Comment by kaapstorm** [comment]
Closing this PR in favor a the series of PRs:
* #29650 
* #29651 
* #29652

**Pull request closed without merging**